x-pack/platform/plugins/shared/fleet/server/routes/cloud_connector/index.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import type { FleetAuthzRouter } from '../../services/security';
+
+import { API_VERSIONS, CLOUD_CONNECTOR_API_ROUTES } from '../../../common/constants';
+import { FLEET_API_PRIVILEGES } from '../../constants/api_privileges';
+
+import { genericErrorResponse } from '../schema/errors';
+
+import { createCloudConnectorHandler, getCloudConnectorsHandler } from './handlers';
+
+export const registerRoutes = (router: FleetAuthzRouter) => {
+  // POST /api/fleet/cloud_connectors
+  router.versioned
+    .post({
+      path: CLOUD_CONNECTOR_API_ROUTES.CREATE_PATTERN,
+      security: {
+        authz: {
+          requiredPrivileges: [
+            {
+              anyRequired: [
+                FLEET_API_PRIVILEGES.AGENT_POLICIES.READ,
+                FLEET_API_PRIVILEGES.INTEGRATIONS.ALL,
+              ],
+            },
+          ],
+        },
+      },
+      summary: 'Create cloud connector',
+      options: {
+        tags: ['oas-tag:Fleet cloud connectors'],
+      },
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.public.v1,
+        validate: {
+          request: {
+            body: schema.object({
```

Comment:
schemas should be defined and imported from `x-pack/platform/plugins/shared/fleet/server/types/rest_spec/`